app/services/host_counter.rb

```diff
@@ -28,14 +28,16 @@ def cache
 
   def counted_hosts
     hosts_scope = Host::Managed.reorder('')
+    grouping = "#{@association}_id"
     case @association.to_s
     when 'organization', 'location'
       # If we are on /organizations or /locations, this allows to display the
       # count for hosts not in the current organization & location.
       hosts_scope = hosts_scope.unscoped
     when 'subnet', 'domain'
       hosts_scope = hosts_scope.joins(:primary_interface)
+      grouping = grouping.prepend("#{Nic::Base.table_name}.")
     end
-    hosts_scope.authorized(:view_hosts).group("#{@association}_id").count
+    hosts_scope.authorized(:view_hosts).group(grouping).count
```

Comment:
Rails treats `Symbol` and `String` differently, would following work?
```suggestion
    hosts_scope.authorized(:view_hosts).group("#{@association}_id".to_sym).count
```

Reply:
```
irb(main):004:0> hosts_scope.authorized(:view_hosts).group("domain_id").count
Traceback (most recent call last):
        2: from lib/tasks/console.rake:5:in `block in <top (required)>'
        1: from (irb):4
ActiveRecord::StatementInvalid (PG::AmbiguousColumn: ERROR:  column reference "domain_id" is ambiguous)
LINE 1: SELECT COUNT(DISTINCT "hosts"."id") AS count_id, domain_id A...
                                                         ^
irb(main):005:0> hosts_scope.authorized(:view_hosts).group("domain_id".to_sym).count
Traceback (most recent call last):
        2: from lib/tasks/console.rake:5:in `block in <top (required)>'
        1: from (irb):5
ActiveRecord::StatementInvalid (PG::AmbiguousColumn: ERROR:  column reference "domain_id" is ambiguous)
LINE 1: SELECT COUNT(DISTINCT "hosts"."id") AS count_id, "domain_id"...
                                                         ^
irb(main):006:0> hosts_scope.authorized(:view_hosts).group("nics.domain_id").count
=> {1=>510}
```